docs/admin/kubeadm-init.md

```diff
@@ -0,0 +1,425 @@
+   `RUNTIME_ENDPOINT` to your own value like `/var/run/{your_runtime}.sock`:
+
+```shell
+  $ cat > /etc/systemd/system/kubelet.service.d/20-cri.conf <<EOF
+Environment="KUBELET_EXTRA_ARGS=--container-runtime=remote --container-runtime-endpoint=$RUNTIME_ENDPOINT --feature-gates=AllAlpha=true"
+EOF
+  $ systemctl daemon-reload
+```
+
+Now `kubelet` is ready to use the specified CRI runtime, and you can continue
+with `kubeadm init` and `kubeadm join` workflow to deploy Kubernetes cluster.
+
+### Using internal IPs in your cluster
+
+In order to set up a cluster where the master and worker nodes communicate with internal IP addresses (instead of publicly addressable ones), you will need to follow these steps.
+
+1. When running init, you must make sure you specify an internal IP for the apiserver's bind address, like so:
+
+   `kubeadm init --apiserver-advertise-address=<private-master-ip>`
+
+2. When a worker node has been provisioned, add a flag to `/etc/systemd/system/kubelet.service.d/10-kubeadm.conf` that specifes the private IP of the worker node:
+
+   `--node-ip=<private-node-ip>`
+
+3. Finally, when you run `kubeadm join` make sure you provide the private IP of the apiserver addressed as defined in step 1.
+
+### Self-hosting the Kubernetes control plane  {#self-hosting}
+As of 1.8, kubeadm can experimentally create a _self-hosted_ Kubernetes control
+plane. This means that key components such as the API server, controller
+manager, and scheduler run as [DaemonSet pods](/docs/concepts/workloads/controllers/daemonset/)
+configured via the Kubernetes API instead of [static pods](/docs/tasks/administer-cluster/static-pod/)
+configured in the kubelet via static files.
+
+**Caution:** Self-hosting is alpha in kubeadm 1.8 but is expected to become the default in
+a future version. To create a self-hosted cluster, pass the `--feature-gates=SelfHosting=true`
+flag to `kubeadm init`.
+{: .caution}
+
+**Warning:** Read carefully current caveats/limitations. 
+{: .warning}
+
+#### Caveats
+
+Kubeadm self-hosting in 1.8 has some important limitations. In particular, a
+self-hosted cluster cannot currently recover from a reboot of the master node
+without manual intervention. This and other limitations are expected to be
+resolved before self-hosting graduates from alpha.
+
+By default, self-hosted control plane pods rely on credentials loaded from
+[`hostPath`](https://kubernetes.io/docs/concepts/storage/volumes/#hostpath)
+volumes. Except for initial creation, these credentials are not managed by
+kubeadm. You can use `--feature-gates=StoreCertsInSecrets=true` to enable an
+experimental mode where control plane credentials are loaded from Secrets
+instead. This requires very careful control over the authentication and
+authorization configuration for your cluster, and may not be appropriate for
+your environment.
+
+In 1.8, the self-hosted portion of the control plane does not include etcd,
```

Comment:
Global comment: Because this is going out with Kubernetes 1.9, we should probably say "kubeadm 1.8", not just "1.8".